### PR TITLE
Use appimage_extract_file_following_symlinks

### PR DESCRIPTION
**Pull request by azubieta**
on the implementation of appimage_register_in_system and appimage_generate_thumbnail

### CODE REVIEW

File: src/libappimage/desktop_integration.c [hunk 1]
```diff
@@ -46,7 +46,12 @@ void traverse_handler_extract_relevant_desktop_integration_files(void* raw_handl
 
         create_parent_dir(target);
 
-        handler->extract_file(handler, entry_data, target);
+        char* link = handler->get_file_link(handler, entry_data);
+        if (link) {
+            appimage_extract_file_following_symlinks(handler->path, file_name, target);
+            free(link);
+        } else
```
Comment (TheAssassin) [inline src/libappimage/desktop_integration.c]: Please use braces on the else clause, too (makes code easier to read and prevents accidental control flow issues).

Comment (azubieta) [reply src/libappimage/desktop_integration.c]: Done.

**Comment by azubieta** [comment]
This PR was meant for libappimage_0.1.x, therefore it's no longer required.
Closing it.

**Comment by TheAssassin** [comment]
This should still go into the 0.1.x branch.

**Comment by azubieta** [comment]
then we should close and re-open the pr but against the 0.1.x branch

**Comment by TheAssassin** [comment]
... you can just change the base branch... can't you?

**Comment by azubieta** [comment]
Wasn't aware of the feature on Github, done.

**Pull request merged**